### PR TITLE
 Build the man page in Cirrus CI 

### PR DESCRIPTION
**Pull request by tobim**

### CODE REVIEW

**Comment by tobim** [comment]
[ch10486]

**Comment by tobim** [comment]
Please don't merge yet, the man page build actually depends on pandoc being availabe in the environment.

File: .ci/scripts/generate-build-tree [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+# e - non-interactive terminals will exit on failures immediately
+# x - print commands before exiting
+set -ex
```
Comment (mavam) [inline .ci/scripts/generate-build-tree]: Is this actually needed for a single-command shell script?

Comment (tobim) [reply .ci/scripts/generate-build-tree]: I removed the `-e` and polished the comment.

**Comment by dominiklohmann** [comment]
You'll need to run `cmake --build ... --target man` manually before package/install.

What about FreeBSD and NixOS? The latter isn't in Cirrus anymore, but we should still provide the man page with these builds if possible.

**Comment by tobim** [comment]
I removed FreeBSD, pandoc is not available pre-built and you don't want to compile that every time.
There is no static build in this repository, so nothing to add for that.

**Comment by tobim** [comment]
> You'll need to run `cmake --build ... --target man` manually before package/install.

I think it will be less surprising if the man page is part of target all.

**Comment by mavam** [comment]
I agree, if pandoc is present, then the user should not have to do anything in addition to build the man page. If it would have a significant effect on build time, I'd be reluctant.

**Pull request merged**